src/main/java/serverutils/pregenerator/filemanager/PregeneratorFileManager.java

```diff
@@ -0,0 +1,101 @@
+package serverutils.pregenerator.filemanager;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import net.minecraft.server.MinecraftServer;
+
+import serverutils.lib.util.misc.PregeneratorCommandInfo;
+import serverutils.pregenerator.filemanager.readwriters.FileReadWriter;
+import serverutils.pregenerator.filemanager.readwriters.SafeFileReadWriter;
+
+public class PregeneratorFileManager {
+
+    private final FileReadWriter commandReadWriter;
+    private final SafeFileReadWriter iterationReadWriter;
+    private final String COMMAND_FOLDER = "pregenerationFiles";
+    private final String COMMAND_FILE = "fileCommand";
+    private final String COMMAND_ITERATION = "fileIteration";
+
+    public PregeneratorFileManager(MinecraftServer server, double xLoc, double zLoc, int radius, int dimensionID)
+            throws IOException {
+        Path temporaryFileSaveFolder = Paths.get("saves").resolve(getWorldFolderPath(server).resolve(COMMAND_FOLDER));
```

Comment:
I would prefer if the temp files are made in some subfolder of the `serverutilities` folder, if they're in the world folder and backups are running then all of the temp files will be zipped up.

Reply:
Makes sense. I'll be sure to move those over. Just had it there for the moment.

Reply:
> I would prefer if the temp files are made in some subfolder of the `serverutilities` folder, if they're in the world folder and backups are running then all of the temp files will be zipped up.

So for pregeneration PR I have made, you mentioned you prefered to have the temp files in the server utilties folder. I think I'll need to make a separate folder in the serverutilties/server folder for each world. For servers this isn't an issue because there's only one world, but for clients that have multiple saves this would be an issue. So I can put it in the serverutilties/server/saves folder and have a folder for each save if you want. I don't know what's best so I thought I'd ask you.

Also, is having it in each backup not fine? If you're doing a pregen and you have to load the backup, you will still need to finish the pregen. Not sure on this.

Reply:
I don't really mind it being in the world folder that much, but right now it's only in there in SP. Servers don't have a `saves` folder.

Reply:
If you want it in the world folder you can use `DimensionManager.getCurrentSaveRootDirectory()` to get the actual root folder.